core/iwasm/common/wasm_memory.c

```diff
@@ -147,23 +147,70 @@ wasm_runtime_free_internal(void *ptr)
 void *
 wasm_runtime_malloc(unsigned int size)
 {
+#if 0
     if (size == 0) {
         LOG_WARNING("warning: wasm_runtime_malloc with size zero\n");
         /* At lease alloc 1 byte to avoid malloc failed */
         size = 1;
     }
 
     return wasm_runtime_malloc_internal(size);
+#endif
+    return wasm_gc_malloc(size);
```

Comment:
Should not change the original implementation, the allocation mode for runtime is initialized by wasm_runtime_init/full_init, it can be initialized as: allocate runtime memory from a global byte buffer, from os_malloc, from user defined malloc function.